notion-server/src/ethereum/mod.rs

```diff
@@ -1,25 +1,122 @@
+#![allow(unused)]
+
+use async_std::sync::{channel, Receiver, Sender};
+
+use snafu::{OptionExt, Snafu};
+
 use std::collections::HashMap;
 
+/// Shorthand for result types returned from the Simulation simulation.
+pub type Result<V, E = Error> = std::result::Result<V, E>;
+
+/// Errors arising from the simulation.
+#[derive(Debug, Snafu)]
+pub enum Error {
+    /// Operation was cancelled because the simulation is shutting down.
+    Terminated,
+}
+
+#[derive(Debug)]
+enum Operation {
+    CreateExecutionEnvironment(args::CreateExecutionEnvironment, Sender<EeIndex>),
+    CreateShardChain(args::CreateShardChain, Sender<u32>),
+    CreateShardBlock(args::CreateShardBlock, Sender<u32>),
+    GetShardBlock(args::GetShardBlock, Sender<ShardBlock>),
+}
+
+#[derive(Debug, Clone)]
+pub struct Handle(Sender<Operation>);
+
+impl Handle {
+    pub async fn create_shard_chain(&self, arg: args::CreateShardChain) -> Result<u32> {
+        let (sender, receiver) = channel(1);
```

Comment:
What is the `1` here?  Is this an arbitrary ID that all the sender/receiver instantiations share to make sure they're all sending/receiving on the same channel?  If not, what does it do, and how do we plan to eventually "hook up" this sender with (what seems to be) the relevant receiver in the `run` method?

Also, is line 36 (and other similar lines in the other functions) just dummy code?  It seems like it would receive operations sent on that channel and then do nothing with them, whereas the `run` function below actually handles them.

Reply:
Nm about the hooking up part -- looks like the "hookup" happens via the instantiation of Handle bc it gets passed the sender when Handle is created by `EthereumSimulation`

Reply:
> What is the `1` here?

It is how many messages are allowed to queue before `.send` blocks.

> Also, is line 36 (and other similar lines in the other functions) just dummy code?

No, this is the actual implementation. The code receives the `Result` on the channel and returns it.

Reply:
Ahhh I see what you did, we're passing the sender from this scope into the struct-level Sender<Operation>, and Operation expects a sender for sending a reply.  Whoops, yep looks good!